api/pkg/s3/bucketput.go

```diff
@@ -75,7 +78,12 @@ func (s *APIService) BucketPut(request *restful.Request, response *restful.Respo
 			return
 		}
 
-		backendName := createBucketConf.LocationConstraint
+		var backendName string
+		if isTier == "True" {
+			backendName = s.getBackendFromTier(ctx, createBucketConf.LocationConstraint)
+		} else {
+			backendName = createBucketConf.LocationConstraint
+		}
 		if backendName != "" {
```

Comment:
better check "" != backendName 
Is trim on the backendName needed ? if yes, please add trim
Please check in other places as well